File: engine/src/main/java/com/terralite/engine/physics/PhysicsComponents.java
package com.terralite.engine.physics;

import com.terralite.engine.entity.ComponentType;

public final class PhysicsComponents {
    public static final ComponentType<Transform> TRANSFORM = ComponentType.of("terralite:transform", Transform.class);
    public static final ComponentType<Transform> PREVIOUS_TRANSFORM = ComponentType.of("terralite:previous_transform", Transform.class);
    public static final ComponentType<Velocity> VELOCITY = ComponentType.of("terralite:velocity", Velocity.class);
    public static final ComponentType<Collider> COLLIDER = ComponentType.of("terralite:collider", Collider.class);

    private PhysicsComponents() {
    }
}

File: engine/src/commonMain/kotlin/com/terralite/engine/physics/PhysicsComponents.kt
package com.terralite.engine.physics

import com.terralite.engine.entity.ComponentType

object PhysicsComponents {
    @JvmField val TRANSFORM: ComponentType<Transform> = ComponentType.of("terralite:transform")
    @JvmField val PREVIOUS_TRANSFORM: ComponentType<Transform> = ComponentType.of("terralite:previous_transform")
    @JvmField val VELOCITY: ComponentType<Velocity> = ComponentType.of("terralite:velocity")
    @JvmField val COLLIDER: ComponentType<Collider> = ComponentType.of("terralite:collider")
    @JvmField val GROUNDED: ComponentType<Boolean> = ComponentType.of("terralite:grounded")
}
